util/llbutil/authprovider/cloudauth/authprovider.go

```diff
@@ -274,6 +283,50 @@ func (ap *authProvider) getAuthConfigForProject(ctx context.Context, org, projec
 	}, nil
 }
 
+type awsAccessKeyCredentials struct {
+	accessKeyID     string
+	secretAccessKey string
+}
+
+// Retrieve implements the CredentialsProvider interface, and only allows for auth based on accessKeyID and secretAccessKey
+func (hc awsAccessKeyCredentials) Retrieve(context.Context) (aws.Credentials, error) {
+	return aws.Credentials{
+		AccessKeyID:     hc.accessKeyID,
+		SecretAccessKey: hc.secretAccessKey,
+	}, nil
+}
+
+func (ap *authProvider) getAuthConfigForProject(ctx context.Context, org, project, host string) (*authConfig, error) {
+	pathPrefix := fmt.Sprintf("std/registry/%s/", host)
```

Comment:
nit: might be nice to move `std/registry` to a const and/or even abstract the paths generation for host,  org/project, `cred_helper`
into their own functions